lib/ansible/modules/cloud/amazon/ec2_eni.py

```diff
@@ -475,26 +473,30 @@ def detach_eni(eni, module):
         module.exit_json(changed=False, interface=get_eni_info(eni))
 
 
-def find_eni(connection, module):
+def uniquely_find_eni(connection, module):
 
     eni_id = module.params.get("eni_id")
-    subnet_id = module.params.get('subnet_id')
     private_ip_address = module.params.get('private_ip_address')
     instance_id = module.params.get('instance_id')
     device_index = module.params.get('device_index')
 
     try:
         filters = {}
-        if subnet_id:
```

Comment:
Basically the idea is that we might want to have multiple nics on the same subnet and maybe create them in an iteration loop (e.g. to attach multiple VMs to a subnet after the instances creation). With the current implementation it is impossible to do.
Removing the `subnet_id` from the filters allow us to fix #19972 requiring explicitely the ip_address or the id to uniquely identify the ENI if we want to update it.